alerting/src/test/kotlin/org/opensearch/alerting/resthandler/MonitorRestApiIT.kt

```diff
@@ -888,4 +894,47 @@ class MonitorRestApiIT : AlertingRestTestCase() {
         val trigger = randomQueryLevelTrigger(actions = listOf(action))
         return randomQueryLevelMonitor(triggers = listOf(trigger))
     }
+
+    @Throws(Exception::class)
+    fun `test search monitors only`() {
+
+        // 1. create monitor
+        val monitor = createRandomMonitor()
+        val createResponse = client().makeRequest("POST", ALERTING_BASE_URI, emptyMap(), monitor.toHttpEntity())
+        assertEquals("Create monitor failed", RestStatus.CREATED, createResponse.restStatus())
+
+        // 2. create destination
+        val chime = Chime("http://abc.com")
+        val destination = Destination(
+            type = DestinationType.CHIME,
+            name = "test",
+            user = randomUser(),
+            lastUpdateTime = Instant.now(),
+            chime = chime,
+            slack = null,
+            customWebhook = null,
+            email = null
+        )
+        val response = client().makeRequest(
+            "POST",
+            LEGACY_OPENDISTRO_DESTINATION_BASE_URI,
+            emptyMap(),
+            destination.toHttpEntity()
+        )
+        assertEquals("Unable to create a new destination", RestStatus.CREATED, response.restStatus())
+
+        // 3. search - must return only monitors.
+        val search = SearchSourceBuilder().query(QueryBuilders.matchAllQuery()).toString()
+        val searchResponse = client().makeRequest(
+            "GET",
+            "$ALERTING_BASE_URI/_search",
+            emptyMap(),
+            NStringEntity(search, ContentType.APPLICATION_JSON)
+        )
+        assertEquals("Search monitor failed", RestStatus.OK, searchResponse.restStatus())
+        val xcp = createParser(XContentType.JSON.xContent(), searchResponse.entity.content)
+        val hits = xcp.map()["hits"]!! as Map<String, Map<String, Any>>
+        val numberDocsFound = hits["total"]?.get("value")
+        assertEquals("Destination objects are also returned by /_search.", 1, numberDocsFound)
```

Comment:
From the integ test, this gets 2 documents

Reply:
I changed to `randomQueryLevelMonitor` util method instead of using `createRandomMonitor`.